test/parallel_api/numeric/numeric.ops/in_place_scan.pass.cpp

```diff
@@ -19,64 +19,532 @@
-  int* excl_input_dev = sycl::malloc_device<int>(10, syclQue);  
-
-  syclQue.memcpy(incl_input_dev, v.data(), v.size()*sizeof(int)).wait();
-  syclQue.memcpy(excl_input_dev, v.data(), v.size()*sizeof(int)).wait();  
-
-  // Inclusive scan (in-place works)
-  inclusive_scan_serial(incl_input_host.begin(),
-                      incl_input_host.end(),
-                      incl_input_host.begin());
-  oneapi::dpl::inclusive_scan(oneapi::dpl::execution::make_device_policy<class TempKernelName1>(syclQue),
-                               incl_input_dev,
-                               incl_input_dev + v.size(),
-                               incl_input_dev );
-  std::vector<int> incl_result_host(v.size());
-  syclQue.memcpy(incl_result_host.data(), incl_input_dev, v.size()*sizeof(int)).wait();
-
-  EXPECT_EQ_RANGES(incl_input_host, incl_result_host, "wrong inclusive_scan results");
-
-  sycl::free(incl_input_dev, syclQue);
-
-  // Exclusive scan (in-place, incorrect results)
-  exclusive_scan_serial( excl_input_host.begin(),
-                       excl_input_host.end(),
-                       excl_input_host.begin(),
-                       0 );
-  oneapi::dpl::exclusive_scan( oneapi::dpl::execution::make_device_policy<class TempKernelName2>(syclQue),
-                               excl_input_dev,
-                               excl_input_dev + v.size(),
-                               excl_input_dev,
-                               0 );
-  std::vector<int> excl_result_host(v.size());
-  syclQue.memcpy(excl_result_host.data(), excl_input_dev, v.size()*sizeof(int)).wait();
-
-  EXPECT_EQ_RANGES(excl_input_host, excl_result_host, "wrong exclusive_scan results");
-
-  sycl::free(excl_input_dev, syclQue);
+    // specialization for hetero policy
+    template <typename Policy, typename Iterator1, typename Size>
+    typename ::std::enable_if<
+        oneapi::dpl::__internal::__is_hetero_execution_policy<typename ::std::decay<Policy>::type>::value &&
+            is_base_of_iterator_category<::std::random_access_iterator_tag, Iterator1>::value,
+        void>::type
+    operator()(Policy&& exec, Iterator1 keys_first, Iterator1 keys_last,
+               Size n)
+    {
+        TestDataTransfer<UDTKind::eKeys, Size> host_keys(*this, n);
+
+        typedef typename ::std::iterator_traits<Iterator1>::value_type KeyT;
+
+        // call algorithm with no optional arguments
+        initialize_data(host_keys.get(), n);
+        update_data(host_keys);
+
+        // copy source host keys state into temp buf
+        typedef typename ::std::iterator_traits<Iterator1>::value_type ValT;
+        const std::vector<ValT> source_host_keys_state(host_keys.get(), host_keys.get() + n);
+
+        auto new_policy = make_new_policy<new_kernel_name<Policy, 0>>(exec);
+        /*auto res1 = */oneapi::dpl::inclusive_scan(new_policy, keys_first, keys_last, keys_first);
+        exec.queue().wait_and_throw();
```

Comment:
`oneapi::dpl::inclusive_scan` is a blocking call.

Reply:
Fixed